### ISSUE TITLE
pd.read_sql crashes on read from Snowflake, version 1.18 and 1.17

### ISSUE THREAD

**Issue by rasnes** [rstudio/reticulate#897]
Hi. I had to downgrade reticulate to version 1.14 to get my setup to work again, it seems like some C stack overflow error or something happening.

Code to reproduce:

```r
library(reticulate)
```

Python:
```python
import pandas as pd
from sqlalchemy import create_engine
from snowflake.sqlalchemy import URL

url = URL(
  #  Snowflake account connection details and credentials
)
engine = create_engine(url)
connection = engine.connect()

query = f"""
  -- some SQL query
"""

df = pd.read_sql_query(query, connection)
```

If I try to run the above in the ´python_repl´, I get `NameError: name '_' is not defined` printed to the screen, and in about 10 seconds later RStudio session crashes.

If I try to knit an RMarkdown document with the above cells, the compilation halts after the python cell and returns:
`Error: C stack usage 17587488537488 is too close to the limit`

I first tried downgrading 1.17 but got the same error, then I "jumped" to 1.14 and it worked again, so I am not sure which version this bug was introduced.

**Comment by kevinushey** [comment]
Do you also see this with the development version of `reticulate`? You can install it with:

```
remotes::install_github("rstudio/reticulate")
```

**Comment by rasnes** [comment]
Heh, I tried now, and it works again on the development version. As an FYI, I tried to look for how to install the development version before trying to downgrade, but since I didn't find anything (I didn't look for that long), I assumed that the development version wasn't available. Perhaps add it under the `Installation` section [here](https://rstudio.github.io/reticulate/), or something, to make it explicit that a dev version is easily available? (Now I learned that I can install the dev version with the standard `remotes::install_github` approach, even though it is not necessarily stated explicitly.)
